stacks/cli/src/commands/login.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  CommandLineOptions,
+  CommandMetadata,
+  CommandMetadataInput,
+  CommandMetadataOption,
+  CommandLineInputs
+} from "@ionic/cli-framework";
+import {Command} from "../interface";
+import * as request from "request-promise-native";
+import * as fs from "fs";
+import {homedir} from "os";
+
+export class LoginCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata<CommandMetadataInput, CommandMetadataOption>> {
+    return {
+      name: "login",
+      summary: "Login to spica with your credentials.",
+      inputs: [
+        {
+          name: "username",
+          summary: "Username of your account"
+        },
+        {
+          name: "password",
+          summary: "Password of your account"
+        }
+      ],
+      options: [
+        {
+          name: "url",
+          summary: "URL of the spica server that will identify your credentials.",
+          type: String,
+          default: "http://localhost:4300"
+        }
+      ]
+    };
+  }
+
+  async run(inputs: CommandLineInputs, options: CommandLineOptions): Promise<void> {
+    const username = inputs[0];
+    const password = inputs[1];
+
+    if (!(username && password)) {
```

Comment:
You should let the required validator to do this check. 

Please check out the other commands to see a similar check.